vaadin-multi-select-list-mixin.html

```diff
@@ -0,0 +1,148 @@
+<!--
+@license
+Copyright (c) 2017 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+<link rel="import" href="vaadin-list-mixin.html">
+<script>
+  window.Vaadin = window.Vaadin || {};
+
+  /**
+   * A mixin for `nav` elements, facilitating multiple selection of childNodes.
+   *
+   * @polymerMixin
+   */
+  Vaadin.MultiSelectListMixin = superClass => class VaadinMultiSelectListMixin extends Vaadin.ListMixin(superClass) {
+    static get properties() {
+      return {
+        /**
+         * Specifies that multiple options can be selected at once.
+         */
+        multiple: {
+          type: Boolean,
+          value: false,
+          reflectToAttribute: true,
+          observer: '_multipleChanged'
+        },
+
+        /**
+         * Array of indexes of the items selected in the items array
+         */
+        selectedValues: {
+          type: Array,
+          notify: true,
+          value: function() {
+            return [];
+          }
+        }
+      };
+    }
+
+    static get observers() {
+      return [
+        `_enhanceMultipleItems(items, multiple, selected, selectedValues, selectedValues.*)`
+      ];
+    }
+
+    ready() {
+      // Should be attached before click listener in list-mixin
+      this.addEventListener('click', e => this._onMultipleClick(e));
+
+      super.ready();
+      this.addEventListener('keydown', e => this._onMultipleKeydown(e));
+    }
+
+    _enhanceMultipleItems(items, multiple, selected, selectedValues) {
+      if (!items || !multiple) {
```

Comment:
- set multiple=true
- select some items
- set multiple=false
- notice that the items are still marked as selected